src/OpenSpaceToolkit/Astrodynamics/Trajectory.cpp

```diff
@@ -244,37 +246,12 @@ Trajectory Trajectory::GroundStrip(
 }
 
 Trajectory Trajectory::GroundStripGeodeticNadir(
-    const trajectory::Orbit& anOrbit, const Array<Instant>& anInstantArray, const Celestial& aCelestial
+    const trajectory::Orbit& anOrbit,
+    [[maybe_unused]] const Array<Instant>& anInstantArray,
+    [[maybe_unused]] const Celestial& aCelestial
 )
 {
-    if (!anOrbit.isDefined())
-    {
-        throw ostk::core::error::runtime::Undefined("Orbit");
-    }
-
-    if (anInstantArray.getSize() < 2)
-    {
-        throw ostk::core::error::RuntimeError("Atleast 2 instants must be provided.");
-    }
-
-    const Shared<Celestial> celestialSPtr = std::make_shared<Celestial>(aCelestial);
-
-    const auto positionGenerator = [&anOrbit, &celestialSPtr](const Instant& anInstant) -> physics::coordinate::Position
-    {
-        const State state = anOrbit.getStateAt(anInstant);
-
-        const LLA lla =
-            LLA::FromPosition(state.getPosition().inFrame(Frame::ITRF(), anInstant), celestialSPtr).onSurface();
-
-        const physics::coordinate::Position position =
-            physics::coordinate::Position::FromLLA(lla, celestialSPtr).inFrame(Frame::GCRF(), anInstant);
-
-        return position;
-    };
-
-    const Array<State> states = computeStates(positionGenerator, anInstantArray);
-
-    return Trajectory(states);
+    [[deprecated("Use Trajectory::GeodeticNadirGroundTrack instead.")]] return Trajectory(Nadir(anOrbit));
```

Comment:
```suggestion
    [[deprecated("Use Trajectory::Nadir instead.")]] return Trajectory(Nadir(anOrbit));
```
(though I think `GeodeticNadirGroundTrack` is more explicit, but up to you - as long as it's consistent)